docs: update Docker instructions across platforms (#315)
aica-technology/api: docs/docs/getting-started/installation/installation-and-launch.md

## Pre-requisites

AICA System software is distributed with Docker images and executed with Docker containers. That is, it can practically
be installed on any system that supports recent Docker versions. However, due to performance advantages (e.g., real-time
task execution), AICA suggests using Linux-based platforms that can be easily configured for such considerations. 

Before proceeding, you will need to install Docker Engine on the host machine. Depending on your platform, the
installation process may vary or require additional steps. Below, you will find instructions per operating system:

<details>
<summary>Linux</summary>

While AICA currently distributes AICA Launcher for Debian-based systems, other distributions where Docker can be
installed work equally as well through any browser. You may find distribution-specific instructions to install Docker
Engine [here](https://docs.docker.com/engine/install/) and follow the manual installation steps to build and run AICA

> Core [here](../../reference/manual-installation-launch.md). Note that you can and it is **recommended** that you skip
> the installation of Docker Desktop for Linux.

Reviewer: ```suggestion
Core [here](../../reference/manual-installation-launch.md). It is **recommended** that you skip the installation of
Docker Desktop for Linux (see below).
```

:::note

For Ubuntu users, make sure to follow
the [post installation steps](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user)
to create the `docker` group and add your user.

:::

:::caution

> For Ubuntu users that need a **Docker Desktop for Linux**, note that its installation does **not** include all the

Reviewer: ```suggestion
For Ubuntu users that need **Docker Desktop for Linux**, note that its installation does **not** include all the
```

Or alternatively
```suggestion
For Ubuntu users that have **Docker Desktop for Linux**, note that its installation does **not** include all the
```

necessary dependencies. Follow the instructions from the provided link to ensure that all requirements are met.

When installed, Docker Desktop for Linux can **not** be used to manage images and containers created by the AICA System. 
Additionally, you may experience some issues when using AICA System software due to Docker Desktop's custom
configuration (see [manual installation](../../reference/manual-installation-launch.md) for more details).

:::

</details>

<details>
<summary>MacOS</summary>

The simplest way to install Docker Engine on MacOS is through 

> [Docker Desktop](https://docs.docker.com/desktop/setup/install/mac-install/). However, some limitations may apply
> regarding commercial use (see relevant article 
> [here](https://www.docker.com/blog/revisiting-docker-hub-policies-prioritizing-developer-experience/)). If you intend to
> operate Docker Desktop within the terms of service, make sure that 
> [these](../../reference/manual-installation-launch.md#configuring-docker-on-your-linux-or-macos-system) additional configuration steps are
> taken.

Reviewer: "If you intend to operate Docker Desktop within the terms of service" -> that's a bit of a weird sentence 😅 

```suggestion
[Docker Desktop](https://docs.docker.com/desktop/setup/install/mac-install/). Ensure that 
[these](../../reference/manual-installation-launch.md#configuring-docker-on-your-linux-or-macos-system) additional configuration steps are
taken.
Note that some limitations may apply regarding commercial use (see relevant article 
[here](https://www.docker.com/blog/revisiting-docker-hub-policies-prioritizing-developer-experience/)).
```

:::note

> The majority of AICA Studio and Launcher features are available in MacOS. However:

Reviewer: ```suggestion
The majority of AICA Studio and Launcher features are available on macOS. However:
```

- Forwarding graphics is currently impossible due to MacOS' incompatibility with OpenGL 2.1+. As a result, tools like
RViz will not be made available through AICA Launcher.
- Realtime applications may be subject to performance limitations due to MacOS' resource management.

:::

</details>

<details>
<summary>Windows</summary>

**No official support is currently provided for Windows machines.** However, with the correct Windows Subsystem for
Linux (WSL) configuration can use the AICA framework manually (see 

> [manual installation instructions](../../reference/manual-installation-launch.md) for reference).

Reviewer 2: Can we have that link as a TODO comment rather? That page doesnt specify any Windows related stuff

Author: let me know if last commit works for you

AICA System software is distributed with Docker images and executed with Docker containers.

> Before proceeding, you will need to install the Docker Engine on the host machine. Depending on your platform, the

Author: ```suggestion
Before proceeding, you will need to install Docker Engine on the host machine. Depending on your platform, the
```

installation process may vary or require additional steps. Below, you will find instructions per operating system:

<details>
<summary>Linux</summary>

> AICA suggests Ubuntu Linux as the main platform, but other distributions where Docker can be installed work equally as 

Author: (point is that since we roll out Launcher for Ubuntu we favour debian-based systems -- i.e., primarily Ubuntu)

well. You may find distribution-specific instructions to install Docker Engine [here](https://docs.docker.com/engine/install/).

> Note that you can and it is **recommended** that you skip the installation of Docker Desktop for Linux.

Author: yup, good point

:::note

For Ubuntu users, make sure to follow

> the [post installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to create the `docker` group

Reviewer 2: We could change this link to https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

:::caution

For Ubuntu users that need a **Docker Desktop for Linux**, note that its installation does **not** include all the
necessary dependencies. Follow the instructions from the provided link to ensure that all requirements are met.

When installed, Docker Desktop for Linux can **not** be used to manage images and containers created by the AICA System. 
Additionally, you may experience some issues when using AICA System software due to Docker Desktop's custom
configuration (see [manual installation](../../reference/manual-installation-launch.md) for more details).

:::

</details>

<details>
<summary>MacOS</summary>

The simplest way to install Docker Engine on MacOS is through 
[Docker Desktop](https://docs.docker.com/desktop/setup/install/mac-install/). However, some limitations may apply
regarding commercial use (see relevant article 
[here](https://www.docker.com/blog/revisiting-docker-hub-policies-prioritizing-developer-experience/)). If you intend to

> operate Docker Desktop within the terms of service, make that 

Reviewer 2: make sure?

> [these](../../reference/manual-installation-launch.md#configuring-docker-desktop) additional configuration steps are

Reviewer 2: ```suggestion
[these](../../reference/manual-installation-launch.md#configuring-docker-on-your-linux-or-macos-system) additional configuration steps are
```

<summary>Windows</summary>

**No official support is currently provided for Windows machines.** However, with the correct Windows Subsystem for

> Linux (WSL) configuration can use the AICA framework manually. 

Reviewer 2: Wait now that I read this sentence again there is the subject missing no?
